QuantLib/ql/termstructures/yield/fittedbonddiscountcurve.cpp

```diff
@@ -180,26 +201,30 @@ namespace QuantLib {
             x = curve_->guessSolution_;
         }
 
-        Simplex simplex(curve_->simplexLambda_);
-        Problem problem(costFunction, constraint, x);
-
-        Natural maxStationaryStateIterations = 100;
-        Real rootEpsilon = curve_->accuracy_;
-        Real functionEpsilon =  curve_->accuracy_;
-        Real gradientNormEpsilon = curve_->accuracy_;
+        if(curve_->maxEvaluations_ > 0){
```

Comment:
I would add some documentation of the behavior. I'd also make more explicit, for instance as:

    if (curve_->maxEvaluations_ == 0) {
        // ...explain...
        return;
    }